src/modules/claims/claims.service.ts

```diff
@@ -269,20 +269,50 @@ export class ClaimsService {
   async registerOnchain(
     claim: Pick<Claim, 'token' | 'subjectAgreement' | 'onChainProof' | 'acceptedBy'>
   ) {
-    if (!readyToBeRegisteredOnchain(claim)) {
+    if (!readyToBeRegisteredOnchain(claim) || !claim.token) {
       throw new Error(ERROR_MESSAGES.CLAIM_WAS_NOT_ISSUED);
     }
-    const { token, subjectAgreement, onChainProof, acceptedBy } = claim;
-    const { claimData, sub } = this._didRegistry.jwt.decode(token) as {
+
+    const { claimData, sub } = this._didRegistry.jwt.decode(claim.token) as {
       claimData: { claimType: string; claimTypeVersion: number };
       sub: string;
     };
+
+    await this.performRegisterOnchain({
+      ...claim,
+      claimType: claimData.claimType,
+      claimTypeVersion: claimData.claimTypeVersion.toString(),
+      subject: sub,
+    });
+  }
+
+  /**
+   * @description Registers issued onchain claim with Claim manager
+   *
+   * @param claimId - id of signed onchain claim
+   */
+  private async performRegisterOnchain(
```

Comment:
I am personally don't see benefits of moving this out of `registerOnChain`. Do you think `performRegisterOnChain` can be used as not part of `registerOnChain`?

Reply:
We not always posses token:
in `request -> issue -> publish` scenario we got token (`registerOnChain`)
in `issue -> publish` scenario we don't have token, only `claimType` and `claimTypeVersion` (`performRegisterOnChain`)

I didn't want to amend registerOnchain as it's public, that's why I overloaded it.  The other option would be to accept `claimType` and `claimTypeVersion` instead of token in all cases. Let me know what you think.

Reply:
sorted - we will decide on breaking changes to API later on  ( registerOnchain)